app/src/main/java/org/simple/clinic/drugs/EditMedicinesModel.kt

```diff
@@ -18,6 +18,14 @@ data class EditMedicinesModel(
     }
   }
 
+  fun isProtocolDrug(prescribedDrug: PrescribedDrug): Boolean {
+    return protocolDrugs!!.any {
+      it.drugs.any { protocolDrug ->
+        protocolDrug.name == prescribedDrug.name && protocolDrug.dosage == prescribedDrug.dosage
+      }
```

Comment:
What do you think of making this block a separate method on the `ProtocolDrugAndDosages` class?

```kotlin
fun matches(prescribedDrug: PrescribedDrug): Boolean {
    return drugs.any { protocolDrug ->
      protocolDrug.name == prescribedDrug.name && protocolDrug.dosage == prescribedDrug.dosage
    }
}
```

Then the `isProtocolDrug` method only operates at its own level of knowledge.

```kotlin
fun isProtocolDrug(prescribedDrug: PrescribedDrug): Boolean {
    return protocolDrugs!!.any { it.matches(prescribedDrug) }
}
```
```

Reply:
Yeah, that makes sense. Will change it

Reply:
Done